cmd/client-s3.go

```diff
@@ -722,67 +730,50 @@ func (c *S3Client) Watch(ctx context.Context, options WatchOptions) (*WatchObjec
 			events = append(events, string(notification.ObjectRemovedAll))
 		case "get":
 			events = append(events, string(notification.ObjectAccessedAll))
+		case "bucket-creation":
+			events = append(events, string(notification.BucketCreatedAll))
+		case "bucket-removal":
+			events = append(events, string(notification.BucketRemovedAll))
```

Comment:
so with `mc mirror --watch site1 site2`, the --watch doesn't seem to wait on events if the sites are in sync.Whereas, it does wait on bucket level mirroring

Reply:
> so with `mc mirror --watch site1 site2`, the --watch doesn't seem to wait on events if the sites are in sync.Whereas, it does wait on bucket level mirroring

I did not understand your point, did you see any bug testing mc ?

Reply:
@vadmeste, I tested this again. --watch mode now seems to be honored and it waits for additional events. However, there is an interesting find - if versioning is enabled on source, the versioning config is not applied to dest bucket. Do you think we should be applying the same bucket metadata config on destination? i.e. if destination supports it , or should we log an error.
```
➜  replication mc ls play/bucket --r --versions
[2020-12-14 13:35:07 PST]    26B null v1 PUT ppp
[2020-12-14 13:35:07 PST]    26B null v1 PUT t2
[2020-12-14 13:35:07 PST]    26B null v1 PUT t22
[2020-12-14 13:35:07 PST]    90B null v1 PUT tax/silver
[2020-12-14 13:35:07 PST]    26B null v1 PUT tax/silver2
➜  replication mc ls dest/bucket --versions --r
[2020-12-14 13:30:15 PST]    26B 8bfc4611-62ec-46b2-b2a5-674dd9e6105a v1 PUT ppp
[2020-12-09 17:31:50 PST]    26B 8eb276f2-36cd-421a-b003-49e878f94d11 v2 PUT t2
[2020-12-07 22:26:20 PST]    26B fcd83bcd-69b3-47d4-a077-7735e7f2c7f2 v1 PUT t2
[2020-12-09 17:33:32 PST]    26B 7fe876cb-ae7f-415b-8ca8-c7429d81f282 v1 PUT t22
[2020-12-14 12:19:05 PST]    90B 5d719010-bae2-4b6c-859e-e98a59a4702f v8 PUT tax/silver
[2020-12-14 12:18:30 PST]    90B 49052ad8-233f-42ce-8df6-97c81ba2a2e6 v7 PUT tax/silver
[2020-12-14 12:17:38 PST]    90B 597310c1-6ae8-47e1-acba-e54ba3c03f80 v6 PUT tax/silver
[2020-12-14 12:15:33 PST]    90B e3868f38-8d9d-4ac7-81d5-4a5b1c58b6e0 v5 PUT tax/silver
[2020-12-14 12:20:39 PST]    26B e3418b27-8167-46b0-af5f-14bceb0893c6 v2 PUT tax/silver2
[2020-12-14 12:19:48 PST]    26B 55d68230-b0fa-4e23-b8d6-585abba37a4c v1 PUT tax/silver2

```

Reply:
> Do you think we should be applying the same bucket metadata config on destination?

I don't think this is needed. mc mirror does not support versioning. So at this point, the user intention is just mirroring last data.

Reply:
ok, in that case. rest of PR looks good.